implementations/rust/c/bindings/src/bindings.rs

```diff
@@ -1035,3 +1030,51 @@ extern "C" {
         attributes: *mut ockam_vault_atecc608a_attributes_t,
     ) -> ockam_error_t;
 }
+#[doc = " @struct  ockam_mutex_dispatch_table"]
+#[doc = " @brief   The Ockam Mutex implementation functions"]
+#[repr(C)]
+#[derive(Debug, Copy, Clone)]
+pub struct ockam_mutex_dispatch_table_t {
+    #[doc = " @brief   Deinitialize the specified ockam mutex implementation."]
+    #[doc = " @param   mutex[in]  The ockam mutex object to deinitialize."]
+    #[doc = " @return  OCKAM_ERROR_NONE on success."]
+    pub deinit:
+        ::core::option::Option<unsafe extern "C" fn(mutex: *mut ockam_mutex_t) -> ockam_error_t>,
+    #[doc = " @brief   Create a mutex"]
+    #[doc = " @param   mutex[in] The ockam mutex implementation to use."]
+    #[doc = " @param   lock[out] Lock object to create."]
+    #[doc = " @return  OCKAM_ERROR_NONE on success."]
+    pub create: ::core::option::Option<
+        unsafe extern "C" fn(
+            mutex: *mut ockam_mutex_t,
+            lock: *mut ockam_mutex_lock_t,
+        ) -> ockam_error_t,
+    >,
+    #[doc = " @brief   Unlock the specified mutex"]
+    #[doc = " @param   mutex[in] The ockam mutex implementation to use."]
+    #[doc = " @param   lock[in]  Lock object to destroy."]
+    #[doc = " @return  OCKAM_ERROR_NONE on success."]
+    pub destroy: ::core::option::Option<
+        unsafe extern "C" fn(mutex: *mut ockam_mutex_t, lock: ockam_mutex_lock_t) -> ockam_error_t,
+    >,
+    #[doc = " @brief   Lock the specified mutex"]
+    #[doc = " @param   mutex[in] The ockam mutex implementation to use."]
+    #[doc = " @param   lock[in]  Lock object to lock."]
+    #[doc = " @return  OCKAM_ERROR_NONE on success."]
+    pub lock: ::core::option::Option<
+        unsafe extern "C" fn(mutex: *mut ockam_mutex_t, lock: ockam_mutex_lock_t) -> ockam_error_t,
+    >,
+    #[doc = " @brief   Unlock the specified mutex"]
+    #[doc = " @param   mutex[in] The ockam mutex implementation to use."]
+    #[doc = " @param   lock[in]  Lock object to unlock."]
+    #[doc = " @return  OCKAM_ERROR_NONE on success."]
+    pub unlock: ::core::option::Option<
+        unsafe extern "C" fn(mutex: *mut ockam_mutex_t, lock: ockam_mutex_lock_t) -> ockam_error_t,
+    >,
+}
+#[repr(C)]
+#[derive(Debug, Copy, Clone)]
+pub struct ockam_mutex_t {
```

Comment:
What are the benefits of this approach over trait objects? This seems rather complex.

Reply:
This is bindings to C object (we use mutex inside C library). Though, it's not used at this moment from rust